addons/pos_restaurant/static/src/overrides/models/pos_store.js

```diff
@@ -319,7 +319,10 @@ patch(PosStore.prototype, {
         );
     },
     tableHasOrders(table) {
-        return Boolean(table.getOrder());
+        return (
+            this.models["pos.order"].filter((o) => o.table_id?.id === table.id && !o.finalized)
+                .length > 0
+        );
```

Comment:
I would like to apply same logic in `getOrder`

Reply:
Sure, now it's 18.0, so we shouldn't remove any method.